dev/clint/src/clint/linter.py

```diff
@@ -244,54 +243,11 @@ def _parse_docstring_args(docstring: str) -> list[str]:
     return args
 
 
-class _ArtifactPathFinder(ast.NodeVisitor):
-    def __init__(self) -> None:
-        self.artifact_path_index: int | None = None
-
-    @classmethod
-    def parse(cls, file: Path) -> int | None:
-        v = cls()
-        v.visit(ast.parse(file.read_text()))
-        return v.artifact_path_index
-
-    def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
-        if node.name == "log_model":
-            artifact_path_index = next(
-                (i for i, arg in enumerate(node.args.args) if arg.arg == "artifact_path"), None
-            )
-            if artifact_path_index is not None:
-                self.artifact_path_index = artifact_path_index
-
-
-@functools.lru_cache(maxsize=32)
-def _find_artifact_path_index(call_path: tuple[str, str]) -> int | None:
-    """
-    Finds the index of the `artifact_path` argument in the function signature of `log_model`.
-    """
-    for p in Path(*call_path).rglob("*.py"):
-        if not p.is_file():
-            continue
-
-        if (idx := _ArtifactPathFinder.parse(p)) is not None:
-            return idx
-    return None
```

Comment:
The main change. We should utilize the index to look up the `artifact_path` parameter.